### PR TITLE
Rework the tutorial level and various small fixes

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR adds the correct textures to the level, introduces basic obstacles, and does some finalizing on it. Also includes some rework in the editor, a complete rework of the level layout and a change of the background color to pitch black.

OPA-41, still misses vent tiles but those are blocked on the implementation of OPA-34 and OPA-102.
Also misses player textures, blocked on OPA-92 for enemies and OPA-116, OPA-118, OPA-119 and OPA-142 for Adam.
Does not include UI textures, partially blocked on OPA-42 and OPA-43, fully blocked on OPA-129, OPA-130 and OPA-131.

### CODE REVIEW

**Review by lokiy999 — APPROVED** [review]
LGTM

File: Assets/Scripts/Core/Tilemap.cs [hunk 1]
```diff
@@ -66,20 +66,29 @@ public class SaveObject {
...
+				DOOR_MIDDLE,
+				DOOR_BOTTOM,
 				COVER,
 			}
 			// Holds the amount of cover this tile gives.
-			public bool cover { get; private set; }
+			public bool cover;
 			// Holds if the tile can be walked over.
-			public bool walkable { get; protected set; }
+			public bool walkable;
```
Comment (Westlanderz) [inline Assets/Scripts/Core/Tilemap.cs:91]: See previous comment.

File: Assets/Scripts/Core/Tilemap.cs [hunk 2]
```diff
@@ -142,21 +151,15 @@ public class SaveObject {
...
 					this.walkable = false;
-					this.hitChanceModifier = pitChanceModifier;
+					this.hitChanceModifier = pitHitChanceModifier;
 					this.cover = false;
 				} else if(nodeSprite == NodeSprite.FLOOR) {
 					this.walkable = true;
 					this.hitChanceModifier = floorHitChanceModifier;
 					this.cover = false;
-				} else if(nodeSprite == NodeSprite.WALL) {
+				} else if(nodeSprite == NodeSprite.WALL_FULL_TEXTURE || nodeSprite == NodeSprite.WALL_BACK_TEXTURE || nodeSprite == NodeSprite.WALL_FULL_MIDDLE_TEXTURE || nodeSprite == NodeSprite.NONE || nodeSprite == NodeSprite.DOOR_TOP || nodeSprite == NodeSprite.DOOR_MIDDLE || nodeSprite == NodeSprite.DOOR_BOTTOM) {
```
Comment (Westlanderz) [inline Assets/Scripts/Core/Tilemap.cs:162]: A bit of a long OR chain, maybe change this to multiple lines for readability.

**Review by Westlanderz — COMMENTED** [review]
See the comments plus why is white_1x1 changing/ moving in the file structure as I added that file already for the health bar system.

File: Assets/Scripts/Core/Tilemap.cs [hunk 3]
```diff
@@ -66,20 +66,29 @@ public class SaveObject {
...
+				WALL_BACK_TEXTURE,
+				// Doors have 3 tiles at the top, actual door and bottom door are prefabs.
+				DOOR_TOP,
+				DOOR_MIDDLE,
+				DOOR_BOTTOM,
 				COVER,
 			}
 			// Holds the amount of cover this tile gives.
-			public bool cover { get; private set; }
+			public bool cover;
```
[the remark this answers is not in the document]
Comment (Dennisbonke) [reply Assets/Scripts/Core/Tilemap.cs:89]: Prefabs need to be able to set this.

File: Assets/Scripts/Core/Tilemap.cs (hunk 1 above)
Comment (Dennisbonke) [reply Assets/Scripts/Core/Tilemap.cs:91]: Prefabs need to be able to set this.

File: Assets/Scripts/Core/Tilemap.cs (hunk 2 above)
Comment (Dennisbonke) [reply Assets/Scripts/Core/Tilemap.cs:162]: Yep, not wrong, I'll split it up.

**Comment by Dennisbonke** [comment]
> See the comments plus why is white_1x1 changing/ moving in the file structure as I added that file already for the health bar system.

It isn't moving, but your file had the file perms 0755 `(rwxr-xr-x)` and I changed them back to the much more sane 0644 `(rw-r--r--)`.

**Comment by Westlanderz** [comment]
> > See the comments plus why is white_1x1 changing/ moving in the file structure as I added that file already for the health bar system.
> 
> It isn't moving, but your file had the file perms 0755 `(rwxr-xr-x)` and I changed them back to the much more sane 0644 `(rw-r--r--)`.

Interesting as either all files by me should have that, or it's because of copying it from code monkey. Either way that's fine then but please check the other files as well, thanks in advance.

**Comment by Dennisbonke** [comment]
> > > See the comments plus why is white_1x1 changing/ moving in the file structure as I added that file already for the health bar system.
> > 
> > 
> > It isn't moving, but your file had the file perms 0755 `(rwxr-xr-x)` and I changed them back to the much more sane 0644 `(rw-r--r--)`.
> 
> Interesting as either all files by me should have that, or it's because of copying it from code monkey. Either way that's fine then but please check the other files as well, thanks in advance.

Did a check of all files in `Assets`, changed everything back to 0644.

**Review by Westlanderz — APPROVED** [review]
LGTM now

**Review by Westlanderz — APPROVED** [review]
LGTM now again

**Pull request merged**